### ISSUE TITLE
Improved, More Explicit Channel Handling in Conda

### ISSUE THREAD

**Issue by jmchilton** [galaxyproject/galaxy#2073]
It sounds like some combinations of channels work and some don't. @bgruening and I went back and forth on this for a while trying to come up with something stable and future proof while maintaining a certain level of abstractness in tool XML files. We converged on to a concept of meta channels (a meta channel is a named set of mutually compatible conda channels).

Add a `config/conda_meta_channels.yml.sample` to `$GALAXY_ROOT/config`.

```
default:
  - bioconda
  - r
  - iuc
  - default

omnia:
  - omnia
  - default

conda-forge:
  - conda-forge
  - default

```

Allow tools to specify a meta_channel hint 

```
   <requirements>
      <conda_hint meta_channel="omnia" /> <!-- default includes bionconda, iuc, and R -->
      <requirement name="package_1" version="1.0" />
      <requirement name="package_2" version="1.0" />
   </requirements>
```

**Comment by mvdbeek** [comment]
I think the meta channel concept is a good idea, but as a tool author I (think I) would like to be able to define these combinations. Would it be possible to ship a conda_meta_channels.yml with a tool?

**Comment by jmchilton** [comment]
One of the key points I guess is to take this out of the tool author's hands - are you sure you want a custom channel - wouldn't you rather contribute to the existing community projects? If we want to specify on the channels on a per-tool basis - I think I'd just toss this meta-channel idea out and let that happen directly.

**Comment by mvdbeek** [comment]
> One of the key points I guess is to take this out of the tool author's hands - are you sure you want a custom channel - wouldn't you rather contribute to the existing community projects?

Definitely, but I'm worried about the situation where my favourite ultra-cool but buggy/deprecated/non-standard configure option clashes with the "official" package, though it may be the only way to get the tool running. This was/is awesome about toolshed packages, you have your own personal namespace, but nothing is stopping you from (preferentially) using iuc/devteam/bgruening/rnateam ... packages.

Or turning it the other way round, if someone changed the `config/conda_meta_channels.yml.sample` file
for whatever reason, we're not necessarily in the same environment, making the debugging more difficult. 

But I can see how shipping / maintaining a `config/conda_meta_channels.yml.sample` with galaxy could be a first start, and if really needed (maybe the scenarios I describe are extreme corner cases!) a per tool channel could still be implemented, right? 
I guess once I have written and fixed a couple of conda recipes I would have a better idea on this issue.

**Comment by jmchilton** [comment]
We just pushed this to 16.10 - @bgruening any updates on this?

**Comment by bgruening** [comment]
The only update I can provide is, that bioconda will be compatible with conda-forge (hopefully) soon. On a related note conda will support channel hierarchies. This means adding a new channel (at the end of the hierarchy chain) will not break everything.
